fbw-a32nx/src/systems/instruments/src/Common/simVars.tsx

```diff
@@ -6,6 +6,112 @@ type SimVarSetter = <T extends SimVarValue>(oldValue: T) => T;
+                    // @ts-ignore
+                    output = new LatLongAltPBH(simvar.getValue_LatLongAltPBH(name, dataSource));
+                } else if (pbhRegex.test(unit)) {
+                    // @ts-ignore
+                    output = new PitchBankHeading(simvar.getValue_PBH(name, dataSource));
+                } else if (pidStructRegex.test(unit)) {
+                    // @ts-ignore
+                    output = new PID_STRUCT(simvar.getValue_PID_STRUCT(name, dataSource));
+                } else if (xyzRegex.test(unit)) {
+                    // @ts-ignore
+                    output = new XYZ(simvar.getValue_XYZ(name, dataSource));
+                } else {
+                    // @ts-ignore
+                    output = simvar.getValueReg(registeredID);
+                }
+            }
+            return output;
+        }
+
+        console.warn(`SimVar handler is not defined (${name})`);
+    } catch (error) {
+        console.warn('ERROR ', error, ` GetSimVarValue ${name} unit : ${unit}`);
+        return null;
+    }
+    return null;
+};
+SimVar.SetSimVarValue = (name, unit, value, dataSource = defaultSource) => {
+    if (value === undefined) {
+        console.warn(`${name} : Trying to set a null value`);
+        return Promise.resolve();
+    }
+    try {
+        // @ts-ignore
+        if (simvar) {
+            const regID = SimVar.GetRegisteredId(name, unit, dataSource);
+            if (regID >= 0) {
+                if (stringRegex.test(unit)) {
+                    return Coherent.call('setValueReg_String', regID, value);
+                }
+                if (boolRegex.test(unit)) {
+                    return Coherent.call('setValueReg_Bool', regID, !!value);
+                }
+                if (numberRegex.test(unit)) {
+                    return Coherent.call('setValueReg_Number', regID, value);
+                }
+                if (latlonaltRegEx.test(unit)) {
+                    return Coherent.call('setValue_LatLongAlt', name, value, dataSource);
+                }
+                if (latlonaltpbhRegex.test(unit)) {
+                    return Coherent.call('setValue_LatLongAltPBH', name, value, dataSource);
+                }
+                if (pbhRegex.test(unit)) {
+                    return Coherent.call('setValue_PBH', name, value, dataSource);
+                }
+                if (pidStructRegex.test(unit)) {
+                    return Coherent.call('setValue_PID_STRUCT', name, value, dataSource);
+                }
+                if (xyzRegex.test(unit)) {
+                    return Coherent.call('setValue_XYZ', name, value, dataSource);
+                }
+
+                return Coherent.call('setValueReg_Number', regID, value);
+            }
+        } else {
+            console.warn('SimVar handler is not defined');
+        }
+    } catch (error) {
+        console.warn(`error SetSimVarValue ${error}`);
+    }
+    return Promise.resolve();
+};
+
```

Comment:
What is the point of adding a typescript version without types? Might as well just use the plain js one... actually this might even be worse as it might not use the .d.ts types?